### PR TITLE
update docker-compose to reflect upstream repo

### PR DESCRIPTION
**Pull request by michaeldelago**
This makes a few changes to `docker-compose.yml` in order to support `azerothcore/azerothcore-wotlk`

this requires 2 pull requests to be merged into the upstream repo:

- https://github.com/azerothcore/azerothcore-wotlk/pull/17530
- https://github.com/azerothcore/azerothcore-wotlk/pull/17529

main changes:
- make worldserver's dependency on client-data be `service_completed_successfully`, since the client data container is now a "job" style container and downloads the client data once
- Use a single volume for client data
    - using 5 volumes isn't necessary
    - the "marker" of which client data version is stored in the datadir, and with 5 separate volumes it wouldn't persist
- let the db-import use its own container
    - the db-import relies on hundreds of megabytes of sql files. 
    - Keeping the sql its own container keeps the worldserver and authserver leaner, and it respects the "single responsibility principle" better
- (proposal) remove the development server from this repo
    - if someone wants to develop on azerothcore with a devcontainer, they should use the one that's in the main repo
    - none of the containers in general ship with source code, and I'm of the opinion that they shouldn't in the first place
    - Naturally, this breaks the dev container in this context, since it requires the source code of azerothcore to function
    - If this repo is in fact scoped for development of azerothcore, I can add it back and ensure basic functionality
    - TODO: update readme to reflect this change

In the mean time of the 3 PRs getting merged, I recommend users look into running their server with the `azerothcore/azerothcore-wotlk` repo. If you can run `docker compose up` with this repo, you can run `docker compose up --build` with that one. the database should seamlessly transfer over. I'd be happy to chat about the process that goes into this (it starts with taking a backup) in the `support-docker` channel in the AzerothCore Discord (@mynameismeat)

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -133,76 +126,10 @@ services:
...
-    cap_add:
-      - SYS_NICE # CAP_SYS_NICE
-    volumes:
-      - ${DOCKER_CLIENT_DATA_FOLDER:-./var/client}:/azerothcore/env/client/Data
-      - ${DOCKER_VOL_TOOLS_CAMERAS:-ac-client-data-cameras}:/azerothcore/env/dist/data/Cameras
-      - ${DOCKER_VOL_TOOLS_DBC:-ac-client-data-dbc}:/azerothcore/env/client/dbc
-      - ${DOCKER_VOL_TOOLS_MAPS:-ac-client-data-maps}:/azerothcore/env/client/maps
-      - ${DOCKER_VOL_TOOLS_VMAPS:-ac-client-data-vmaps}:/azerothcore/env/client/vmaps
-      - ${DOCKER_VOL_TOOLS_MMAPS:-ac-client-data-mmaps}:/azerothcore/env/client/mmaps
-    profiles: [dev]
```
Comment (Helias) [inline docker-compose.yml:205]: why are you removing all these ac-dev*?

File: .github/workflows/test-install.yml [hunk 2]
```diff
@@ -38,29 +38,3 @@ jobs:
...
-
-      # TODO: fix after this is solved: https://github.com/docker/compose/issues/9515#issuecomment-1303538917
-      - name: Test
-        shell: bash
-        run: |
-          cp .github/docker-compose.dev-server.yml docker-compose.override.yml
-          cp .github/config.sh config.sh
-          docker compose up -d ac-dev-database
-          docker compose run --no-deps ac-dev-server bash -c "./acore.sh compiler build"
-          docker compose run --no-deps ac-dev-server bash -c "./env/dist/bin/dbimport"
```
Comment (Helias) [inline .github/workflows/test-install.yml:66]: why did you remove this? for incompatibility reasons?

Comment (michaeldelago) [reply .github/workflows/test-install.yml:66]: yes. If the dev resources are removed from the repo, it means that tests for them will fail

File: docker-compose.yml (hunk 1 above)
Comment (michaeldelago) [reply docker-compose.yml:205]: I noted in the PR's body that containers related to development have been removed in favor of those in `azerothcore-wotlk`

Having 2 repositories organized for development on azerothcore makes it more difficult to provide support for both. Because of that, I think it's in our best interest to tighten the scope of this repository as much as possible.

**Comment by zls3434** [comment]
i used this repo, but i got an error like this:
```
acore-docker-ac-db-import-1     | ERROR 1146 (42S02) at line 2 in file: '/azerothcore/data/sql/updates/db_world/2023_04_24_00.sql': Table 'acore_world.version' doesn't exist
acore-docker-ac-db-import-1     | 
acore-docker-ac-db-import-1     | Applying of file '/azerothcore/data/sql/updates/db_world/2023_04_24_00.sql' to database 'acore_world' failed! If you are a user, please pull the latest revision from the repository. Also make sure you have not applied any of the databases with your sql client. You cannot use auto-update system and import sql files from AzerothCore repository with your sql client. If you are a developer, please fix your sql query.
acore-docker-ac-db-import-1     | Could not update the World database, see log for details.

```

**Comment by csmlngjd99** [comment]
hello i use this branch  gave rise to the same problem as this :
https://github.com/azerothcore/acore-docker/issues/37

**Comment by smoove** [comment]
Using this PR results in:

```
acore-docker-ac-db-import-1     | ERROR 1146 (42S02) at line 2 in file: '/azerothcore/data/sql/updates/db_world/2023_04_24_00.sql': Table 'acore_world.version' doesn't exist
```

Edit: It worked after i started the database container first (docker compose up ac-database-1 or whatever) and then only started the db import container also (docker compose up ac-db-import-1 or whatever).

After doing this once, i can now reliably run the entire thing via docker compose up

**Comment by Helias** [comment]
I did not encounter any issues so far, so I merged this PR
